statistics-dashboard-service/src/main/java/eu/europeana/statistics/dashboard/service/StatisticsService.java

```diff
@@ -114,56 +131,77 @@ public FilteringResult queryDataWithFilters(StatisticsFilteringRequest statistic
     List<StatisticsData> queries = new ArrayList<>();
 
     // Get all Value fields type
-    List<MongoStatisticsField> filterMongoStatisticFields = Arrays.stream(MongoStatisticsField.values()).filter(
-        field -> field != MongoStatisticsField.UPDATED_DATE && field != MongoStatisticsField.CREATED_DATE
+    List<MongoStatisticsField> filterMongoStatisticFields = Arrays.stream(
+        MongoStatisticsField.values()).filter(
+        field -> field != MongoStatisticsField.UPDATED_DATE
+            && field != MongoStatisticsField.CREATED_DATE
             && field != MongoStatisticsField.DATASET_ID).collect(Collectors.toUnmodifiableList());
 
     // Execute a breakdown query for each Value field
-    filterMongoStatisticFields.forEach(field -> queries.add(query.withBreakdowns(field).queryForStatistics()));
+    filterMongoStatisticFields.forEach(
+        field -> queries.add(query.withBreakdowns(field).queryForStatistics()));
 
     return queries;
   }
 
-  private StatisticsQuery prepareFilteringQuery(StatisticsFilteringRequest statisticsRequest)
+  private StatisticsQuery prepareFilteringQuery(FiltersWrapper statisticsRequest)
       throws BreakdownDeclarationFailException {
+    StatisticsFilteringRequest filters = statisticsRequest.getFilters();
     StatisticsQuery query = mongoSDDao.createStatisticsQuery();
 
-    Map<MongoStatisticsField, Set<String>> parsedValueFilters = RequestUtils.parseValuesFiltersFromRequest(statisticsRequest);
-    Map<MongoStatisticsField, ValueRange> parsedRangeFilters = RequestUtils.parseRangeFiltersFromRequest(statisticsRequest);
-    List<MongoStatisticsField> breakdowns = RequestUtils.parseBreakdownsFromRequest(statisticsRequest);
+    Map<MongoStatisticsField, Set<String>> parsedValueFilters = RequestUtils.parseValuesFiltersFromRequest(
+        filters);
+    Map<MongoStatisticsField, ValueRange> parsedRangeFilters = RequestUtils.parseRangeFiltersFromRequest(
+        filters);
+    List<MongoStatisticsField> breakdowns = RequestUtils.parseBreakdownsFromRequest(filters);
 
     // Set each parsed value into query
     parsedValueFilters.forEach(query::withValueFilter);
-    parsedRangeFilters.forEach((key, value) -> query.withRangeFilter(key, value.getFrom(), value.getTo()));
+    parsedRangeFilters.forEach(
+        (key, value) -> query.withRangeFilter(key, value.getFrom(), value.getTo()));
     query.withBreakdowns(breakdowns.toArray(MongoStatisticsField[]::new));
 
+    List<String> datasetIdValues = statisticsRequest.getFilters().getDatasetId().getValues();
```

Comment:
This should be ideally handled from RequestUtils.parseValuesFiltersFromRequest call. That means that the eu.europeana.statistics.dashboard.common.iternal.MongoStatisticsField#DATASET_ID should be updated accordingly. In that way the code block here can be omitted completely.

Reply:
I have made the necessary changes and removed this block of code thanks to the previous comment's changes